apis/run/v1alpha3/clusterbootstraptemplate_types.go

```diff
@@ -50,16 +52,46 @@ type ClusterBootstrapPackage struct {
 	ValuesFrom *ValuesFrom `json:"valuesFrom,omitempty"`
 }
 
+// ValuesFrom specifies how values for package install are retrieved from
+// +kubebuilder:object:generate=false
 type ValuesFrom struct {
 	// +optional
-	Inline string `json:"inline,omitempty"`
+	// +kubebuilder:validation:Schemaless
+	// +kubebuilder:validation:Type=object
+	// +kubebuilder:pruning:PreserveUnknownFields
+	Inline map[string]interface{} `json:"inline,omitempty"`
 	// +optional
 	SecretRef string `json:"secretRef,omitempty"`
 	// +optional
 	ProviderRef *corev1.TypedLocalObjectReference `json:"providerRef,omitempty"`
 }
 
-//+kubebuilder:object:root=true
+// DeepCopyInto is a deepcopy function, copying the receiver, writing into out. in must be non-nil.
```

Comment:
question: trying to understand why the deep copy func gets generated here instead of `zz_generated.deepcopy.go` ?

Reply:
These are custom definitions added similar to OSImage type
(Refer to comment by @vijaykatam => https://github.com/vmware-tanzu/tanzu-framework/issues/1694#issuecomment-1059422094)